src/lib/Libdb/db_postgres_impl.c

```diff
@@ -815,6 +815,53 @@ pbs_db_destroy_connection(pbs_db_conn_t *conn)
 int
 pbs_db_save_obj(pbs_db_conn_t *conn, pbs_db_obj_info_t *obj, int savetype)
 {
+	pbs_db_attr_list_t *cache_attrs = NULL;
+	pbs_db_attr_list_t *db_attrs = NULL;
+
+	switch (obj->pbs_db_obj_type) {
+		case PBS_DB_SVR:
+				cache_attrs = &((pbs_db_svr_info_t *) obj->pbs_db_un.pbs_db_svr)->cache_attr_list;
+				db_attrs    = &((pbs_db_svr_info_t *) obj->pbs_db_un.pbs_db_svr)->db_attr_list;
+				break;
+
+		case PBS_DB_SCHED:
+				cache_attrs = &((pbs_db_sched_info_t *) obj->pbs_db_un.pbs_db_sched)->cache_attr_list;
+				db_attrs    = &((pbs_db_sched_info_t *) obj->pbs_db_un.pbs_db_sched)->db_attr_list;
+				break;
+
+		case PBS_DB_QUEUE:
+				cache_attrs = &((pbs_db_que_info_t *) obj->pbs_db_un.pbs_db_que)->cache_attr_list;
+				db_attrs    = &((pbs_db_que_info_t *) obj->pbs_db_un.pbs_db_que)->db_attr_list;
+				break;
+
+		case PBS_DB_NODE:
+				cache_attrs = &((pbs_db_node_info_t *) obj->pbs_db_un.pbs_db_node)->cache_attr_list;
+				db_attrs    = &((pbs_db_node_info_t *) obj->pbs_db_un.pbs_db_node)->db_attr_list;
+				break;
+
+		case PBS_DB_MOMINFO_TIME:
+				break;
+
+		case PBS_DB_JOB:
+				cache_attrs = &((pbs_db_job_info_t *) obj->pbs_db_un.pbs_db_job)->cache_attr_list;
+				db_attrs    = &((pbs_db_job_info_t *) obj->pbs_db_un.pbs_db_job)->db_attr_list;
+				break;
+
+		case PBS_DB_JOBSCR:
+				break;
+
+		case PBS_DB_RESV:
+				cache_attrs = &((pbs_db_resv_info_t *) obj->pbs_db_un.pbs_db_resv)->cache_attr_list;
+				db_attrs    = &((pbs_db_resv_info_t *) obj->pbs_db_un.pbs_db_resv)->db_attr_list;
+				break;
+	}
+
+	/*  for postgres, implement distributed cache by saving to database */
+	if (cache_attrs && db_attrs && (cache_attrs->attr_count > 0)) {
+		list_move(&cache_attrs->attrs, &db_attrs->attrs);
```

Comment:
Just wanted to know whether we are adding cache_attrs on top of db_attrs here so that both are saved to database. Also as we are going to change this logic when distributed cache is implemented can we have this piece of code in a separate function which will make things easier in future.

Reply:
yes, so we are merging the two separate lists (which happens only in case of multiserver) to one list for the postgres implementation. So cache_attrs being moved to db_attrs (with a simple few link changes). This will change for aerospike implemnetation where aerospike provides a inbuilt caching mechanism. It could also change, if we decide to add a dist cache support like memcached. 

We don't know what the interface will look like, so making a function will not help now.